tomli/_re.py

```diff
@@ -12,8 +12,26 @@
 # - 00:32:00
 _TIME_RE_STR = r"([01][0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9])(?:\.([0-9]{1,6})[0-9]*)?"
 
-RE_NUMBER = re.compile(
-    r"""
+
+@lru_cache(maxsize=None)
```

Comment:
Is `lru_cache` needed here since Python caches regex operations anyway?

https://github.com/python/cpython/blob/3.10/Lib/re.py#L285-L312

Reply:
Very good point! Yeah we can probably simplify by storing the non-compiled regex strings here, and then use
```python
import re
re.match(NUMBER_RE_STR, src, pos)
```
in `_parser.py`. I assume this is what you meant?

Thanks!

Reply:
I guess the only potential downside is that the CPython cache has a max size so these may get invalidated, and that I assume this is a CPython implementation detail that we then rely on.

Reply:
Yes. Sorry, if I was a bit unclear, but this project is quite new to me. I've just seen a lot of projects complicating things unnecessarily and juggling regexes in the name of "optimization" even though Python probably cached everything like they wanted without any extra effort in `re.match()`, `re.sub()`, and so on. :)

Reply:
Cool! And yeah I 100% agree with your observation. Initially there was no optimization really, but I just prefer the API / readability of using compiled regexes vs regex strings. But that leads to eager compilation which then proved to be an issue in ofek's use case.

Reply:
Ah okay there is actually an issue with using regex strings. The following is not valid Python:
```python
re.match(NUMBER_RE_STR, src, pos)
```
This is incorrect because the `pos` argument is not available in `re.match`, only `re.Pattern.match`...